### ISSUE TITLE
need support product grouping in core

### ISSUE THREAD

**Issue by pindge** [opendatacube/datacube-core#1359]
`ASTERS` are not part of DEA products, but due to alphabetical ordering they are always the first products showing up on query `dc.list_products()`

![image](https://user-images.githubusercontent.com/24644475/203658194-98e9688b-70f5-4e8f-a0f3-882dcdf12a20.png)

core should add a feature to support pre-identified grouping, ie.

`dc.list_products(group='c3')`

**Comment by robbibt** [comment]
Yep, being able to sort our datasets by the "Baseline satellite data", "Inland water" etc groupings would be really nice!

**Comment by pindge** [comment]
one potential solution is to add `product_category` to product metadata definition, and to query against it 

> [8:38 AM] Paul Haesler
    If product_category was listed in the metadata_type as a search field, you could use dc.index.products.search_robust(product_category="foo")  - but you'd still need to trawl through the results manually.  search_robust is the only method provided for searching products, and it's really intended for internal use by the dataset search methods rather than direct user use.

> Even then, it's highly inefficient - there's an implicit assumption that the product table is small enough that it's OK to do a full table scan every query, so honestly, you may as well just do that manually.
​[8:40 AM] Paul Haesler
    The search_robust code would look something like:

> `[ prod for prod, unmatchable in dc.index.search_robust(product_category="foo") if not unmatchable]`

**Comment by SpacemanPaul** [comment]
As noted in Teams, a products version of `search_by_metadata` would be doable.

**Comment by SpacemanPaul** [comment]
Oh correction to above:
```
[ prod for prod, unmatchable in dc.index.products.search_robust(product_category="foo") if not unmatchable]
```

is equivalent to:

```
dc.index.products.search(product_category="foo")
```

But again, this requires product_category to be configured search field.
